tests/conftest.py

```diff
@@ -234,43 +259,55 @@ def inner(
             sleep(5)
 
         if select_duo_push and test_env == 'eval':
-            wait = WebDriverWait(current_browser, 10)
-            wait.until(EC.element_to_be_clickable((By.XPATH, "//a[contains(text(), 'Other options')]")))
-            current_browser.wait_for_tag('a', 'Other options').click()
-            current_browser.wait_for_tag('b', 'Other options to log in')
+            duo_push(current_browser)
 
         if test_env == 'eval':
+            # focus on the correct element based on if we are retrying the passcode,
+            # or its the first time we enter the passcode
             wait = WebDriverWait(current_browser, 10)
-            if assert_success and retry:
+            if retry:
                 element = wait.until(EC.element_to_be_clickable((By.XPATH,
                                                                  "//input[contains(@id, 'passcode-input')]")))
             else:
-                element = wait.until(EC.visibility_of_element_located((By.XPATH, "//div[contains(text(), 'Bypass code') "
-                                                                                 "and "
-                                                                                 "contains( "
-                                                                       "@class, 'row') and contains(@class, "
-                                                                                 "'display-flex') ]"))
-                                 )
+                element = wait.until(EC.visibility_of_element_located((By.XPATH,
+                                                                       "//div[contains(text(), 'Bypass code') and "
+                                                                       "contains(@class, 'row') and contains(@class, "
+                                                                       "'display-flex')]")))
+
             current_browser.snap()
             element.click()
             current_browser.snap()
-            if assert_success and retry:
-                current_browser.execute_script("arguments[0].value = '';", element)
-                current_browser.snap()
+            if retry:
+                clear_passcode(current_browser, element)
             current_browser.send_inputs(passcode)
             current_browser.snap()
-            current_browser.snap()
             current_browser.wait_for_tag('button', 'Verify').click()
+            sleep(5)
```

Comment:
I understand that the async "wait" related stuff isn't perfect. Given that can you either

1. try using wait here if you haven't yet
2. add a comment indicating what's being waited for with the sleep

Reply:
removed